apollo-ios/Sources/Apollo/MaxRetryInterceptor.swift

```diff
@@ -51,11 +99,47 @@ public class MaxRetryInterceptor: ApolloInterceptor {
     }
     
     self.hitCount += 1
-    chain.proceedAsync(
-      request: request,
-      response: response,
-      interceptor: self,
-      completion: completion
-    )
+    
+    // Apply exponential backoff delay if enabled and this is a retry (hitCount > 1)
+    if self.configuration.enableExponentialBackoff && self.hitCount > 1 {
+      let delay = calculateExponentialBackoffDelay()
+      DispatchQueue.main.asyncAfter(deadline: .now() + delay) { [weak self] in
+        guard let self = self else { return }
+        chain.proceedAsync(
+          request: request,
+          response: response,
+          interceptor: self,
+          completion: completion
+        )
+      }
+    } else {
+      chain.proceedAsync(
+        request: request,
+        response: response,
+        interceptor: self,
+        completion: completion
+      )
+    }
+  }
+  
+  /// Calculates the exponential backoff delay based on current retry attempt.
+  ///
+  /// - Returns: The calculated delay in seconds, including jitter if enabled.
+  private func calculateExponentialBackoffDelay() -> TimeInterval {
```

Comment:
Might be a good idea to have tests for the calculated backoff delays since this has the potential to wreck the default interceptor chain.